supportedBackends/htcondor/src/main/scala/cromwell/backend/impl/htcondor/CondorJobExecutionActor.scala

```diff
@@ -0,0 +1,195 @@
+
+
+  val runtimeAttributes = {
+    val evaluateAttrs = call.task.runtimeAttributes.attrs mapValues (_.evaluate(lookup, callEngineFunction))
+    // Fail the call if runtime attributes can't be evaluated
+    TryUtil.sequenceMap(evaluateAttrs, "Runtime attributes evaluation").get.mapValues(_.valueString)
+  }
+
+  /**
+    * Restart or resume a previously-started job.
+    */
+  override def recover: Future[BackendJobExecutionResponse] = Future.failed(new Exception("HtCondorBackend currently doesn't support recovering jobs."))
+
+  /**
+    * Execute a new job.
+    */
+  override def execute: Future[BackendJobExecutionResponse] = Future(executeTask())
+
+  /**
+    * Abort a running job.
+    */
+  override def abortJob(): Unit = Future.failed(new Exception("HtCondorBackend currently doesn't support aborting jobs."))
+
+  override def preStart(): Unit = {
+    log.debug(s"Creating execution folder: $executionDir")
+    executionDir.toString.toFile.createIfNotExists(true)
+    try {
+      val localizedInputs = localizeInputs(jobPaths, docker = false, fileSystems, jobDescriptor.inputs)
+      val command = call.task.instantiateCommand(localizedInputs, callEngineFunction, identity).get
+      log.debug(s"Creating bash script for executing command: $command.")
+      //TODO: need to access other requirements for submit file from runtime requirements
+      val attributes = Map(HtCondorRuntimeKeys.executable -> command,
+        HtCondorRuntimeKeys.output -> s"$stdoutPath",
+        HtCondorRuntimeKeys.error -> s"$stderrPath")
+      val condorSubmitFile = cmds.submitCommand(submitPath, attributes)
+      writeScript(condorSubmitFile, executionDir)
+    } catch {
+      case ex: Exception =>
+        log.error(ex, s"Failed to prepare task: ${ex.getMessage}")
+    }
+  }
+
+  /**
+    * Writes the script file containing the user's command from the WDL as well
+    * as some extra shell code for monitoring jobs
+    */
+  private def writeScript(instantiatedCommand: String, containerRoot: Path) = {
+    jobPaths.script.write(
+      s"""#!/bin/sh
+          |cd $containerRoot
+          |$instantiatedCommand
+          |echo $$? > rc
+          |""".stripMargin)
+  }
+
+  private def executeTask(): BackendJobExecutionResponse = {
+    val process = extProcess.externalProcess(argv, ProcessLogger(stdoutWriter.writeWithNewline, stderrWriter.writeWithNewline))
+    val processReturnCode = process.exitValue() // blocks until process (i.e. condor submission) finishes
+    log.debug(s"processReturnCode  : $processReturnCode")
+
+    List(stdoutWriter.writer, stderrWriter.writer).foreach(_.flushAndClose())
+    log.debug(s"done flushing")
+    val stderrFileLength = Try(Files.size(stderrPath)).getOrElse(0L)
+    log.debug(s"stderr file length : $stderrFileLength ")
+
+    val rc = Try(returnCodePath.contentAsString.stripLineEnd.toInt)
+    log.debug(s"received return code: $rc ")
+
+    if (processReturnCode != 0) {
```

Comment:
This if else block is really bad. I will try to find a better way to rewrite this. Functionality first 😓